kong/db/schema/init.lua

```diff
@@ -1703,6 +1705,28 @@ local function collect_field_reference(refs, key, reference)
 end
 
 
+local function validate_deprecation_exclusiveness(data, shorthand_value, shorthand_name, shorthand_definition)
+  if shorthand_definition.deprecation and shorthand_definition.deprecation.replaced_with then
+    for _, replaced_with_element in ipairs(shorthand_definition.deprecation.replaced_with) do
+      local new_field_value = replaced_with_element.translation and replaced_with_element.translation(data)
```

Comment:
IMO, we could simply reject if both the new field and the deprecated field are present. Is it necessary to allow redundancy?

Reply:
We need to allow for redundancy for backward compatilibity. decK currently dumps both new and old fields. Actually since the changes requested back in January there's currently no way to get API response without deprecated fields - so both of these are saved to decK dump. Then when it's sent it transfers both new and old fields.

Reply:
Should we avoid doing the translation twice? It seems we do it one more time for migration

Reply:
I'm not sure I follow 🤔 Are you asking whether we can skip migrations?

Reply:
> I'm not sure I follow 🤔 Are you asking whether we can skip migrations?

I mean we will translate the field once for the config compatibility, but here we do it again for the comparing purpose. Could we save one time for the translation?

Reply:
We potentially could but I think in order to do that cleanly it would require **a lot(!)** of code refactoring. Let's just go with this right now and then let's refactor `process_auto_fields` because it became a bit of monstrosity 😰 
It's a good idea though - to go through that once!